djmoney/models/fields.py

```diff
@@ -294,6 +294,8 @@ def deconstruct(self):
             del kwargs["default"]
         else:
             kwargs["default"] = self.default.amount
+        if self.default_currency is None:  # None needs to be preserved
+            kwargs["default_currency"] = self.default_currency
```

Comment:
I could, alternately, do:
```
kwargs["default_currency"] = self.default_currency
...
if self.default_currency is not None and self.default_currency != DEFAULT_CURRENCY:
    kwargs["default_currency"] = str(self.default_currency)
```

Reply:
Am not sure I understand the benefits of this? Why is `str()` necessary here and not in the alternative solution?

Reply:
`str()` is the original code, and my guess is that it's there to cast Currency objects. If we were to coerce `None`, it'd say `"None"` in the DB, as if it was currency.

I realize now that it's not clear from looking at the line I commented on, but take a look at the whole segment to get a feel for it.

Reply:
If it's preferable to coerce to a string, then that should be the solution? I understand that it shouldn't be done for `None` :)

Reply:
Just to clarify, that == the current version?

Reply:
Ah no, I was sensing that you preferred your own alternative suggestion and not the current PR's solution? Could you maybe clarify which one of them you prefer?

Reply:
None, hence the question :) Let's keep the one that's currently used then.

But, **do not merge this yet**. It might be the cause of inability to migrate to a nullable currency field. I _think_ it's tied to my specific usecase (I've already migrated non-nullable currency and am trying to use this fix to make it nullable), but I can't be sure.

It essentially changes the migration state mid-migration and makes the migration executor think that the currency field was already nullable. It's absolutely cursed.

I'm preparing an investigation report I'll be posting as soon as I'm a bit clearer on the cause.

Reply:
Maybe it's better to make a test case then?

Reply:
From what I'm seeing, yes. No clue how to best do it, though, since it _might_ need to be at the SQL level, _or_ assert that the resulting db fields are actually nullable.

Reply:
> assert that the resulting db fields are actually nullable.

I think that sounds like a good test-case that doesn't need to introspect the database.

Reply:
You could add a new field/model here: https://github.com/django-money/django-money/blob/master/tests/testapp/models.py

Reply:
Will do, though we should probably move out of this review thread. I'll resolve it now, let's continue on the PR  itself.